ldes-server-pagination/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/pagination/batch/PaginationJobDefinitions.java

```diff
@@ -8,55 +8,65 @@
 import org.springframework.core.task.TaskExecutor;
 import org.springframework.transaction.PlatformTransactionManager;
 
+import java.util.List;
+
 @Configuration
 public class PaginationJobDefinitions {
 	public static final String PAGINATION_JOB = "pagination";
 	public static final String NEW_VIEW_PAGINATION_JOB = "newViewPagination";
+	private static final int CHUNK_SIZE = 1000;
 
 	@Bean
 	public Step paginationStep(JobRepository jobRepository, PlatformTransactionManager transactionManager,
 	                           Partitioner bucketPartitioner, ItemReader<Page> pageItemReader,
-	                           ItemProcessor<Page, Page> pageRelationsProcessor,
-	                           ItemWriter<Page> memberAssigner,
-	                           TaskExecutor taskExecutor) {
+	                           ItemProcessor<Page, List<Page>> pageRelationsProcessor,
+	                           ItemWriter<List<Page>> memberAssigner,
+	                           @Qualifier("asynch") TaskExecutor taskExecutor) {
 		return new StepBuilder("paginationMasterStep", jobRepository)
 				.partitioner("memberBucketPartitionStep", bucketPartitioner)
 				.step(new StepBuilder("paginationStep", jobRepository)
-						.<Page, Page>chunk(150, transactionManager)
+						.<Page, List<Page>>chunk(CHUNK_SIZE, transactionManager)
 						.reader(pageItemReader)
 						.processor(pageRelationsProcessor)
 						.writer(memberAssigner)
 						.allowStartIfComplete(true)
+						.taskExecutor(taskExecutor)
 						.build()
 				)
-				.taskExecutor(taskExecutor)
 				.allowStartIfComplete(true)
 				.build();
 	}
 
 	@Bean
 	public Step newViewPaginationStep(JobRepository jobRepository, PlatformTransactionManager transactionManager,
 	                                  Partitioner bucketPartitioner, ItemReader<Page> pageItemReader,
-	                                  ItemProcessor<Page, Page> pageRelationsProcessor,
-	                                  ItemWriter<Page> memberAssigner,
-	                                  TaskExecutor taskExecutor) {
+	                                  ItemProcessor<Page, List<Page>> pageRelationsProcessor,
+	                                  ItemWriter<List<Page>> memberAssigner,
+	                                  @Qualifier("asynch") TaskExecutor taskExecutor) {
 		return new StepBuilder("newViewPaginationMasterStep", jobRepository)
 				.partitioner("memberBucketPartitionStep", bucketPartitioner)
 				.step(new StepBuilder("paginationStep", jobRepository)
-						.<Page, Page>chunk(150, transactionManager)
+						.<Page, List<Page>>chunk(CHUNK_SIZE, transactionManager)
 						.reader(pageItemReader)
 						.processor(pageRelationsProcessor)
 						.writer(memberAssigner)
 						.allowStartIfComplete(true)
+						.taskExecutor(taskExecutor)
 						.build()
 				)
-				.taskExecutor(taskExecutor)
 				.allowStartIfComplete(true)
 				.build();
 	}
+
+	@Bean("asynch")
+	public TaskExecutor taskExecutor() {
```

Comment:
Maybe the bean could be named more to the context, e.g. pagianationTaskExecutor perhaps?